src/games/strategy/triplea/settings/ValueRange.java

```diff
@@ -0,0 +1,23 @@
+package games.strategy.triplea.settings;
+
+/**
+ * Encapsulates the permissible range for the value of a specific setting.
+ *
+ * <p>
+ * Implementations may choose to track other associated values in addition to the range itself, such as a default value.
+ * </p>
+ */
+public interface ValueRange {
```

Comment:
Good point.  Let me explain the thought process that got me here.

I first considered making the new `SettingInputComponent#getValueRange()` method return `Optional<IntegerValueRange>`.  However, that smelled bad to me considering that the type of many settings is not an integer--it just looked weird seeing <tt><b>Integer</b>ValueRange</tt> used for a generic setting.

Obviously, it doesn't make sense to define a range for settings that have types such as `String`.  However, I thought of a valid use case where a setting could have type `Float` or `Double`, and a range would make sense for those types.  Now that I could imagine another range type such as `DoubleValueRange`, I didn't want to put the responsibility for handling the different range types in `SettingsWindow#createInputValueRangeDescription()`, so I decided to do it polymorphically by adding the `ValueRange#getDescription()` method.

To specifically answer your question about the value this interface offers currently, the answer is nothing other than removing the awkwardness of associating an optional **integer** value range with every setting.  I don't feel strongly enough about that smell to argue for keeping the new interface; I'm ok with backing out the new interface and just using `IntegerValueRange` as is.

One other solution to consider, if not keeping `ValueRange`, is to change the new `SettingInputComponent` method:

```java
Optional<ValueRange> getValueRange();
```

to

```java
Optional<String> getValueRangeDescription();
```

And put the responsibility for formatting the value range description back in the `SettingInputComponentFactory#buildIntegerText()` method, where it was originally.  This might be the most practical solution as it addresses your YAGNI concern and removes the code smell I pointed out above.